plugins/rbac-backend/src/service/role-manager.test.ts

```diff
@@ -511,6 +929,52 @@ describe('BackstageRoleManager', () => {
       );
     });
 
+    // user:default/mike should inherits role from group:default/team-a, but we have cycle dependency: team-a -> team-c.
+    // So return false on call hasLink.
+    //
+    //     Hierarchy:
+    //
+    // group:default/team-a -> role:default/team-a   group:default/team-b
+    //       ↓       ↑                                    ↓
+    // group:default/team-c -> role:default/team-c   group:default/team-d
```

Comment:
You didn't use  role:default/team-c, so you can remove it from Hierarchy, or really use it with roleManager.addLink('group:default/team-c', 'role:default/team-c');